cs3api4lab/api/cs3_file_api.py

```diff
@@ -112,11 +102,20 @@ def read_file(self, file_path, endpoint=None):
         file_get = None
         try:
             protocol = [p for p in init_file_download.protocols if p.protocol == "simple"][0]
-            headers = {
-                'x-access-token': self.auth.authenticate(),
-                'X-Reva-Transfer': protocol.token    # needed if the downloads pass through the data gateway in reva
-            }
-            file_get = requests.get(url=protocol.download_endpoint, headers=headers)
+            if protocol.opaque and init_file_download.protocols[0].opaque.map['webdav-file-path'].value:
```

Comment:
Hi, I added some comments in the code, hopefully this will be more clear. So yes, after I spoke to Ishank it turns out that when reading files shared in OCM shares, the reva service returns webdav endpoint and token to read the file.